user/user_test.go

```diff
@@ -0,0 +1,22 @@
+package user
+
+import "testing"
+
+// TestEmail_Validate ensures email validation functions as expected.
+func TestEmail_Validate(t *testing.T) {
+	tests := []struct {
+		email string
+		valid bool
+	}{
+		{"valid@example.com", true},
+		{"valid@nebulous.tech", true},
+		{"invalid_nebulous.tech", false},
+		{"", false},
```

Comment:
useful additions https://gist.github.com/cjaoude/fd9910626629b53c4d25

Reply:
Wow, some of these are really terrible. I'm kind of torn about this. On one hand, I'd like to allow all valid email addresses. On the other, I don't want to allow invalid ones. Even the author's own suggested regex doesn't catch the odd ones and that ultimate regex he links to doesn't work out of the box in Go and I don't think it's worth it to invest the time to fix and escape it.

OTOH, if your email really is along the lines of `very.”(),:;<>[]”.VERY.”very@\ "very”.unusual@strange.example.com` then maybe not being serviced is kind of on you at that point? :)

There's also the question of why is `email@example.web` invalid.

Anyway, thanks a lot for the suggestion, it was a very fun rabbit hole to dive into! I think I'll add some of the sane suggestions but for the moment I don't think I'll change the regex. Unless someone offers a better one, of course.

Reply:
> There's also the question of why is email@example.web invalid.

`.web` isn't a valid TLD i believe was the answer there. 

yea, some of those are really really wild haha